apiplugin/file/filehost.go

```diff
@@ -298,13 +298,18 @@ func (f *fileSvcImpl) read(ctx context.Context, req *file.ReadRequest, resp *fil
 	}
 
 	myFileInfo := fileDataCache[fid]
-	fpath := myFileInfo.path
 	// check file status, we cannot read a closed file
 	if myFileInfo.status == Fs_Close {
-		pcontext.Errorf(ctx, "file is closed, cannot be read: %s", fpath)
+		pcontext.Errorf(ctx, "file is closed, cannot be read: %d", fid)
 
 		return int32(file.FileErr_FileClosedError)
 	}
+	// check reader existitance, we cannot read a file without reader
+	if myFileInfo.reader == nil {
```

Comment:
After thinking about it, I think you should change the error text and the name of the error to something that doesn't reference the reader.  I would suggest that maybe this should be an "internal error" because your code should not put the `reader` field in a bad state.  I assume that you don't intentionally leave a fileInfo with a nil reader, right?